shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory.Builder;
+
+/*
+ * Class that modifies Store entries in datastore.
+ */
+public class StoreDatastoreHandler {
+    
+    private String id;
+    private Key key;
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    public StoreDatastoreHandler(String id) {
+        this.id = id;
+        this.key = new Builder("Store", id).getKey();
+    }
+
+    /*
+     * Wrapper that handles cases where store is in datastore or not
+     */
+    public void placeStore(Map<String, String[]> ratingsMap) {
+        try {
+            //Found store in datastore
+            Entity existingStoreEntity = datastore.get(key);
+            this.updateStore(existingStoreEntity, ratingsMap); 
+        } catch (Exception EntityNotFoundException) {
+            //Store not in datastore, so create entry
+            this.newStore(ratingsMap);
+        }
+    }
+
+    /*
+     * Creates Rating Entity, does not put in datastore
+     */
+    private Entity createRatingsEntity(Map<String, String[]> ratingsMap) {
+        Entity ratingEntity = new Entity("Rating");
+        ratingsMap.forEach((String ratingField, String[] ratingValue) -> {
+            
+            /* http request getParamMap method formats in array instead of
+             * single value.
+             */
+            ratingEntity.setProperty(ratingField, Double.parseDouble(ratingValue[0]));
+        });
+        return ratingEntity;
+    }
+
+    /*
+     * Places new store along with rating
+     */
+    private void newStore(Map<String, String[]> ratingsMap) {
+        
```

Comment:
Could get rid of the extra spaces for styling

Reply:
I agree with Gabe, you should probably fix the styling and add a little more commenting before merging :)